src/coherence/ai.py

```diff
@@ -337,6 +337,87 @@ def __init__(self, extractor: Union[ValueExtractor[T, E], str], over_sampling_fa
         self.oversamplingFactor = over_sampling_factor
 
 
+@proxy("coherence.hnsw.HnswIndex")
+class HnswIndex(AbstractEvolvable):
+    DEFAULT_SPACE_NAME: Final[str] = "COSINE"
+    """The default index space name."""
+    DEFAULT_MAX_ELEMENTS: Final[int] = 4096
+    """
+    The default maximum number of elements the index can contain is 4096
+    but the index will grow automatically by doubling its capacity until it
+    reaches approximately 8m elements, at which point it will grow by 50%
+    whenever it gets full.
+    """
```

Comment:
Could we add spaces between the definitions and their docs to improve readability?

Reply:
Added spaces